src/services/codefixes/fixMissingTypeAnnotationOnExports.ts

```diff
@@ -0,0 +1,1192 @@
+        // No inline assertions on binding patterns
+        if (findAncestor(targetNode, isBindingPattern)) {
+            return undefined;
+        }
+
+        // No inline assertions on enum members
+        if (findAncestor(targetNode, isEnumMember)) {
+            return undefined;
+        }
+        // No support for typeof in extends clauses
+        if (isExpressionTarget && findAncestor(targetNode, isHeritageClause)) {
+            return undefined;
+        }
+        // Can't inline type spread elements. Whatever you do isolated declarations will not infer from them
+        if (isSpreadElement(targetNode)) {
+            return undefined;
+        }
+
+        const variableDeclaration = findAncestor(targetNode, isVariableDeclaration);
+        const type = variableDeclaration && typeChecker.getTypeAtLocation(variableDeclaration);
+        // We can't use typeof un an unique symbol. Would result in either
+        // const s = Symbol("") as unique symbol
+        // const s = Symbol("") as typeof s
+        // both of which are not correct
+        if (type && type.flags & TypeFlags.UniqueESSymbol) {
+            return undefined;
+        }
+
+        if (!(isExpressionTarget || isShorthandPropertyAssignmentTarget)) return undefined;
+
+        const { typeNode, mutatedTarget } = inferType(targetNode, type);
+        if (!typeNode || mutatedTarget) return undefined;
+
+        if (isShorthandPropertyAssignmentTarget) {
+            changeTracker.insertNodeAt(
+                sourceFile,
+                targetNode.end,
+                createAsExpression(
+                    getSynthesizedDeepClone(targetNode.name),
+                    typeNode,
+                ),
+                {
+                    prefix: ": ",
+                },
+            );
+        }
+        else if (isExpressionTarget) {
+            changeTracker.replaceNode(
+                sourceFile,
+                targetNode,
+                createSatisfiesAsExpression(
+                    getSynthesizedDeepClone(targetNode),
+                    typeNode,
+                ),
+            );
+        }
+        else {
+            Debug.assertNever(targetNode);
+        }
+        return [Diagnostics.Add_satisfies_and_an_inline_type_assertion_with_0, typeToStringForDiag(typeNode)];
+    }
+
+    function suggestVariableName(node: Node) {
+        const nameParts: string[] = [];
+        while (!(isVariableDeclaration(node) || isPropertyDeclaration(node) || isStatement(node))) {
+            if (isPropertyAssignment(node)) {
+                const propName = node.name;
+                addPropertyName(propName);
+            }
+            node = node.parent;
+        }
+        if ((isVariableDeclaration(node) || isPropertyDeclaration(node)) && !isBindingPattern(node.name)) {
+            addPropertyName(node.name);
+        }
+        return nameParts.filter(s => isIdentifierText(s, program.getCompilerOptions().target)).reverse().join("_");

```

Comment:
I might be missing other instances of this issue: anything that appears in `computedOptions` in `src/compiler/utilities.ts` needs to be accessed through a helper function:

```suggestion
        return nameParts.filter(s => isIdentifierText(s, getEmitScriptTarget(program.getCompilerOptions()))).reverse().join("_");
```

Reply:
Done